purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1252,10 +1253,9 @@ class Purchases internal constructor(
             })
     }
 
-    private val handler = Handler(Looper.getMainLooper())
     private fun dispatch(action: () -> Unit) {
         if (Thread.currentThread() != Looper.getMainLooper().thread) {
-            handler.post(action)
+            mainHandler.post(action)
```

Comment:
I thought about adding a test for this, but we would have to refactor this a bit to be able to test it (Like the if condition above, we would need to mock the `Looper.getMainLooper()` method). I thought it was better to get the fix out the door as soon as possible.